### ISSUE TITLE
Logout Implementation

### ISSUE THREAD

**Issue by saikiranvadlakonda** [auth0/react-native-lock#36]
Hi All,
We have integrated this in mobile app development, But we didn't find implementation for logout. That is how to call logout on lock object. And we have gone through auth0-lock.js and other files, we didn't find any code for logout. Could you guys tell us how to logout a user from auth0 using react-native-lock.

**Comment by hzalaz** [comment]
@saikiranvadlakonda this implementation of Lock doesn't keep any session nor stores the tokens, it's up to the library user to handle them, and that's why there is no `logout` method.

**Comment by saikiranvadlakonda** [comment]
@hzalaz  As per the login flow, auth0 asks user to choose any one of the identity provider, assume user selected google(Gmail) and then prompts for emailid and password, after successful login it continues to our app, up to here it's fine, but whenever user logs out from app user is not logging out from auth0 and google(Gmail). How to logout a user from Gmail also.

**Comment by hzalaz** [comment]
It's Google that's still logged in not Auth0, you still login against Auth0 every time. I have in our pipeline to patch the libraries to avoid the other IdP to keep the session whenever possible.

**Comment by saikiranvadlakonda** [comment]
Hi @hzalaz ,
I want to know how to logout a user from google whenever he logs out from app. 
We can observe that when user clicks on G button which invokes google login, there it is not asking user to enter username and password, because of old login is still available and it is taking that login and continuing. Is there any possible to logout user from google.

I'm attaching a pic which is displayed whenever user clicks on G button and We would like know about that '_login with google'_ dialog box, we manually want to invoke that and logout a user.
Awaiting for response.
![img-20160504-wa0005](https://cloud.githubusercontent.com/assets/14762284/15011296/b08ec688-120e-11e6-92cc-0da50f6b169b.jpg)


Thanking you

**Comment by hzalaz** [comment]
Sorry for the delay, we'll add a parameter when invoking google to force the account chooser

**Comment by asashay** [comment]
Hi!

Having same issue in my app.
@hzalaz have you made any enhancements to deal with this issue?

@saikiranvadlakonda did you come with the solution for this? Will be very helpful...

Cheers

**Comment by septemptus** [comment]
@hzalaz has this enhancement progressed in any way? I have a similar issue -- I am creating an application that uses Instagram login, and once I log in once the Instagram session is maintained and I cannot log out / switch to another account unless I revoke the app from authorized apps list on instagram.

**Comment by hzalaz** [comment]
No, we have other priorities right now and the fix we were planning to add works only for google since it let's you pick the account to login. For other IdP  clearing the webview session might be the only way to achieve this until we provide support for using the phone's browser.

**Comment by septemptus** [comment]
@hzalaz thanks, I'll look into clearing the webview session for now then

**Comment by maksim-harakhovich-itechart** [comment]
@septemptus did you succeed? Can you please share how to do that?

**Comment by septemptus** [comment]
@maksim-harakhovich-itechart I have moved away from auth0 for my project for now, but I believe [react-native-cookie](https://github.com/joeferraro/react-native-cookies) package can solve this issue if you clear the cookies.

**Comment by maksim-harakhovich-itechart** [comment]
@septemptus unfortunately it doesn't help :(

**Comment by stami** [comment]
The logout functionality is really needed! Can we get this issue an another look?

**Comment by abelaska** [comment]
No to be able to pick a different Google account after user logs out from application is really stupid.

Please help us and implement the feature. Thank you.

**Comment by MSSPL-PiyaleeMaiti** [comment]
@hzalaz , I am using react-native-lock. I am facing same issue for facebook login. Login credential store from first time login, after logout from app it's not removed. Can you suggest any solution for logout from facebook when logout from app.

**Comment by sheprdinc** [comment]
I'm using facebook login with Auth0... logging out works fine, except it's not a true log out... it's a partial log out where only the local session is removed from AsyncStorage, however, facebook continues to remember the user, which makes it impossible for a different facebook user to login on the same device, unless you ask your user to delete the app and re-install, which is not an acceptable thing to ask users to do, especially at scale, and is unfortunately forcing me to re-think Auth0 and consider other options, after so much has been invested already to make Auth0 part of our stack.

@hzalaz i understand what you are saying, but for many of us, clearing the webview session seems impossible or alien in React Native. Perhaps more possible in Swift/Obj-c, but not in pure React Native... and since this Auth0 Lock is like a Swift/Obj-c overlay, it just seems like it makes a lot more sense for Lock to handle this, than for React Native developers. In fact, this is a service/add-on our team would gladly pay for and/or subscribe to.

@maksim-harakhovich-itechart @stami @abelaska @MSSPL-PiyaleeMaiti have any of you arrived at a solution? Because I also tested out clearing the webview session with 'react-native-cookie' library, but that did not work. I will keep trying work-arounds as well, and if I figure it out, I will post the solution here. Good luck everyone!

**Comment by Msspl-PrashenjeetRoy** [comment]
@sheprdinc We have not getting solution till now. It's now a being a big issue for us. Facebook and linkedin both now not logout. Before few days Linkedin was working fine and we can login and logout easily but now Linkedin also creating issue like facebook.
